staging/src/k8s.io/client-go/tools/cache/reflector_test.go

```diff
@@ -358,6 +360,88 @@ func TestReflectorListAndWatchWithErrors(t *testing.T) {
 	}
 }
 
+func TestReflectorListAndWatchInitConnBackoff(t *testing.T) {
+	maxBackoff := 50 * time.Millisecond
+	table := []struct {
+		numConnFails  int
+		expLowerBound time.Duration
+		expUpperBound time.Duration
+	}{
+		{5, 32 * time.Millisecond, 64 * time.Millisecond}, // case where maxBackoff is not hit, time should grow exponentially
+		{40, 35 * 2 * maxBackoff, 40 * 2 * maxBackoff},    // case where maxBoff is hit, backoff time should flatten
+
+	}
+	for _, test := range table {
+		t.Run(fmt.Sprintf("%d connection failures takes at least %d ms", test.numConnFails, 1<<test.numConnFails),
+			func(t *testing.T) {
+				stopCh := make(chan struct{})
+				connFails := test.numConnFails
+				fakeClock := clock.NewFakeClock(time.Unix(0, 0))
+				bm := wait.NewExponentialBackoffManager(time.Millisecond, maxBackoff, 100*time.Millisecond, 2.0, 1.0, fakeClock)
+				done := make(chan struct{})
+				go func() {
+					i := 0
+					for {
+						select {
+						case <-done:
+							return
+						default:
+							if fakeClock.HasWaiters() {
+								step := (1 << (i + 1)) * time.Millisecond
+								if step > maxBackoff*2 {
+									step = maxBackoff * 2
+								}
+								fakeClock.Step(step)
+								i++
+							}
+							time.Sleep(100 * time.Microsecond)
```

Comment:
This looks great, I personally would probably not put so much in the default: case, maybe leave it empty[*]. Then this chunk of code isn't indented so much. But that's just stylistic.

[*] alternatively, put the time.Sleep into the default case, and move the select statement below the if HasWaiters check.

Reply:
done.